tests/apollo/test_skvbc_checkpoints.py

```diff
@@ -128,6 +126,32 @@ async def test_checkpoint_propagation_after_crashed_replicas_comeback(self, bft_
         # verify checkpoint propagation to stale node after it comes back up
         self.assertEqual(checkpoint_after_stale, checkpoint_after_primary)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_continuous_checkpoint_creation_while_replicas_crash(self, bft_network):
+        """
+        Here we continuously create checkpoints while replicas continuously crash and come back.
+        1) Given a BFT network, we make sure all nodes are up
+        2) Crash f replicas, excluding the current primary
+        3) Send sufficient number of client requests to trigger checkpoint protocol
+        4) Make sure checkpoint is created
+        5) Bring the f replicas up
+        6) Repeat steps 2 - 5 nb_crashes number of times
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        n = bft_network.config.n
+        f = bft_network.config.f
+        c = bft_network.config.c
+
+        self.assertEqual(len(bft_network.procs), n,
+                         "Make sure all replicas are up initially.")
+        current_primary = await bft_network.get_current_primary()
+
+        await self._create_checkpoints_while_crashing(bft_network, skvbc, f, c,
+                                                      current_primary, nb_crashes=2)
+
     async def _crash_replicas_without_primary(
```

Comment:
As discussed in a previous PR, let's refactor this using the already existing utility @yontyon introduced some time ago:
https://github.com/vmware/concord-bft/blob/master/tests/apollo/util/bft.py#L399

Reply:
@teoparvanov, updated.